services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchFlowUtils.java

```diff
@@ -0,0 +1,155 @@
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.switchmanager;
+
+import static java.util.Collections.singletonList;
+import static org.projectfloodlight.openflow.protocol.OFVersion.OF_13;
+
+import net.floodlightcontroller.core.IOFSwitch;
+import net.floodlightcontroller.util.FlowModUtils;
+import org.projectfloodlight.openflow.protocol.OFFactory;
+import org.projectfloodlight.openflow.protocol.OFFlowMod;
+import org.projectfloodlight.openflow.protocol.OFMeterFlags;
+import org.projectfloodlight.openflow.protocol.OFMeterMod;
+import org.projectfloodlight.openflow.protocol.OFMeterModCommand;
+import org.projectfloodlight.openflow.protocol.action.OFAction;
+import org.projectfloodlight.openflow.protocol.action.OFActions;
+import org.projectfloodlight.openflow.protocol.meterband.OFMeterBandDrop;
+import org.projectfloodlight.openflow.protocol.oxm.OFOxms;
+import org.projectfloodlight.openflow.types.DatapathId;
+import org.projectfloodlight.openflow.types.MacAddress;
+import org.projectfloodlight.openflow.types.OFBufferId;
+import org.projectfloodlight.openflow.types.OFPort;
+import org.projectfloodlight.openflow.types.TableId;
+import org.projectfloodlight.openflow.types.U64;
+
+import java.util.Arrays;
+import java.util.Set;
+
+/**
+ * Utils for switch flow generation.
+ */
+public final class SwitchFlowUtils {
+
+    /**
+     * OVS software switch manufacturer constant value.
+     */
+    public static final String OVS_MANUFACTURER = "Nicira, Inc.";
+
+    private SwitchFlowUtils() {
+    }
+
+    /**
+     * Create a MAC address based on the DPID.
+     *
+     * @param dpId switch object
+     * @return {@link MacAddress}
+     */
+    public static MacAddress convertDpIdToMac(DatapathId dpId) {
+        return MacAddress.of(Arrays.copyOfRange(dpId.getBytes(), 2, 8));
+    }
+
+    /**
+     * Create sent to controller OpenFlow action.
+     *
+     * @param sw switch representation
+     * @return OpenFlow Action
+     */
+    public static OFAction actionSendToController(final IOFSwitch sw) {
+        OFActions actions = sw.getOFFactory().actions();
+        return actions.buildOutput().setMaxLen(0xFFffFFff).setPort(OFPort.CONTROLLER)
+                .build();
+    }
+
+    /**
+     * Create set destination MAC address OpenFlow action.
+     *
+     * @param sw switch representation
+     * @param macAddress MAC address to set
+     * @return OpenFlow Action
+     */
+    public static OFAction actionSetDstMac(final IOFSwitch sw, final MacAddress macAddress) {
+        OFOxms oxms = sw.getOFFactory().oxms();
+        OFActions actions = sw.getOFFactory().actions();
```

Comment:
From whole IOFSwitch object you use only factory field. Maybe we should use factory in method args (as it is `Utils` method)

Reply:
Done